### DISCUSSION TITLE
Xiaomi Philips Zhirui Bedside Lamp connecyion issue

### DISCUSSION THREAD

**Discussion by jokerigno** [rytilahti/python-miio#1894] [General]
Hi,

I have 2 Xiaomi Philips Zhirui Bedside Lamps and I noticed that they randomly become unavailable in Home Assistant while they both still work with physical switch. The only solution is to power cycle. There's no a specific patters: sometimes are both, sometimes just one, sometimes happens weekly and someothers are ok for months.

Both are connected to an UNIFI AP (the same), no other device have same issue. 

Any hint?

**Comment by rytilahti** [comment]
So there are devices that have a watchdog that restarts a device when they are not able to form a connection with the cloud service, but yours is first report about completely losing the local access.

Alas, there is probably pretty much nothing that this library can do in such cases. If your device is blocked from the internet, you could try to allow access to some services (ntp, dns, ..) to see if that helps.

**Reply by jokerigno** [reply]
i spent some hours making some test and yes: if i reopen internet access to the lamp it stop being unresponsive (even if lag seems still high).
Fun fact is that this issue only occurs in a bedlamp (the other still works as expected).
